### ISSUE TITLE
Stuck on search field when using CEC remote (Android TV)

### ISSUE THREAD

**Issue by peat80** [rumboalla/apkupdater#407]
When you focus the search field on tv you are trapped inside and cannot escape using remote control. You can only press back to exit the app, but on reopening the search field is focussed again which makes the app unuseable. Only clear app data / reinstalling does help.

**Comment by peat80** [comment]
Also the sliders for columns / frequency are not moveable wit the Remote.

**Comment by peat80** [comment]
No issues while using touch control on the smartphone.

**Comment by rumboalla** [comment]
You can't exit after clicking on the search icon?

**Comment by peat80** [comment]
No, if you enter something with remote you only can move the cursor on the entered text but not exit the text field. After pressing back and reopening you are trapped in an empty text field with blinking cursor and can not do anything.

**Comment by peat80** [comment]
On input text the left / right button moves the cursor one character left / right and up / down will move cursor on the start / end of the inserted text but not outside the textbox if that helps. 🤔

**Comment by rumboalla** [comment]
> On input text the left / right button moves the cursor one character left / right and up / down will move cursor on the start / end of the inserted text but not outside the textbox if that helps. 🤔

I'll have to test this on a proper TV, moving up and down on the emulator works as intended.

**Comment by peat80** [comment]
This seems to be some strange cec issue. Tried the apk on my bravia tv and exiting the text box works fine. 🧐

It also works on the shield tv when using the original remote. 🙄

It works not when using bravia tv remote or harmony remote via cec on shield tv. 🤔

**Comment by rumboalla** [comment]
> This seems to be some strange cec issue. Tried the apk on my bravia tv and exiting the text box works fine. 🧐
> 
> It also works on the shield tv when using the original remote. 🙄
> 
> It works not when using bravia tv remote or harmony remote via cec on shield tv. 🤔

Not sure what to do about this 🤔
The slider thing is definitely a bug with no easy solution. I may have to change the sliders for a dropdown or textfield on TV.

**Comment by peat80** [comment]
The search thing is a really strange one. Probably not easy to fix if at all. But atleast i know now that the original remote works and search is also not used that often. 🤔😀

The 2 sliders for the colums maybe should just be hidden when the tv ui is used. They are not needed.

And the notification alarm time could be changed to a textfield maybe. This should be useful to set on tv too. 😀

While we are at it, it would be cool to add a setting that there is only a notification shown when there are updates in a future version. The "no updates found" notification is a little bit more annoying than useful. 😁

**Comment by teohhanhui** [comment]
FWIW the search field bug occurs with Xiaomi TV's stock remote control. It's bad because sometimes once you launch ApkUpdater app you're already stuck in the text field with no way out.

Even pairing a Bluetooth mouse doesn't help because when I click on "Install now" the app just exits (this second issue might not have anything to do with ApkUpdater though...)

**Comment by peat80** [comment]
Tested behaviour now again. Using latest ci-build.
Seems to work now.

Closing as fixed.
